presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/ElasticsearchClient.java

```diff
@@ -454,4 +464,51 @@ public int compare(String left, String right)
             return rightLength - leftLength;
         }
     }
+
+    static TransportClient createTransportClient(ElasticsearchConnectorConfig config, TransportAddress address)
+    {
+        return createTransportClient(config, address, Optional.empty());
+    }
+
+    static TransportClient createTransportClient(ElasticsearchConnectorConfig config, TransportAddress address, Optional<String> clusterName)
+    {
+        Settings settings;
+        Builder builder;
+        TransportClient client;
+        if (clusterName.isPresent()) {
+            builder = Settings.builder()
+                    .put("cluster.name", clusterName.get());
+        }
+        else {
+            builder = Settings.builder()
+                    .put("client.transport.ignore_cluster_name", true);
+        }
+        switch (config.getCertificateFormat()) {
+            case PEM:
+                settings = builder
+                        .put(SEARCHGUARD_SSL_TRANSPORT_PEMCERT_FILEPATH, config.getPemcertFilepath())
+                        .put(SEARCHGUARD_SSL_TRANSPORT_PEMKEY_FILEPATH, config.getPemkeyFilepath())
+                        .put(SEARCHGUARD_SSL_TRANSPORT_PEMKEY_PASSWORD, config.getPemkeyPassword())
+                        .put(SEARCHGUARD_SSL_TRANSPORT_PEMTRUSTEDCAS_FILEPATH, config.getPemtrustedcasFilepath())
+                        .put(SEARCHGUARD_SSL_TRANSPORT_ENFORCE_HOSTNAME_VERIFICATION, false)
+                        .build();
+                client = new PreBuiltTransportClient(settings, SearchGuardSSLPlugin.class).addTransportAddress(address);
+                break;
+            case JKS:
+                settings = Settings.builder()
```

Comment:
@zhenxiao Is this intended to drop `client.transport.ignore_cluster_name` or `cluster.name` that are set above?

Reply:
whoops, I might lose the context, client.transport.ignore_cluster_name is set to true, when cluster.name is not configured. This is for one use case, where it is a virtual ELK cluster with a number of underlying ELK clusters, sharing search nodes. Not able to configure cluster name in this case.